src/main/java/ca/warp7/frc2023/subsystems/Elevator.java

```diff
@@ -0,0 +1,121 @@
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxAlternateEncoder;
+import com.revrobotics.SparkMaxLimitSwitch;
+import com.revrobotics.SparkMaxLimitSwitch.Type;
+import com.revrobotics.SparkMaxPIDController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Elevator extends SubsystemBase {
+    private static Elevator instance;
+
+    // returns the instance of the subsystem
+    public static Elevator getInstance() {
+        if (instance == null) instance = new Elevator();
+        return instance;
+    }
+
+    public static CANSparkMax createMasterSparkMAX(int deviceID) {
+        CANSparkMax master = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        master.restoreFactoryDefaults();
+        master.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        master.enableVoltageCompensation(12.0);
+        return master;
+    }
+
+    public CANSparkMax createFollowerSparkMAX(int deviceID) {
+        CANSparkMax follower = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        follower.restoreFactoryDefaults();
+        follower.follow(LeftMain);
+        follower.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        follower.enableVoltageCompensation(12.0);
+        return follower;
+    }
+
+    // creates an inverted SparkMAx follower motor
+    public CANSparkMax createFollowerSparkMAXInverted(int deviceID) {
+        CANSparkMax follower = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        follower.restoreFactoryDefaults();
+        follower.follow(LeftMain, true);
+        follower.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        follower.enableVoltageCompensation(12.0);
+        return follower;
+    }
+
+    // makes a leader and a follower motor for the left side
+    public CANSparkMax LeftMain = createMasterSparkMAX(kElevatorLeftMain);
+    public CANSparkMax LeftSecondary = createFollowerSparkMAX(kElevatorLeftSecondary);
+
+    // makes the inverted follower motors for the right side
+    public CANSparkMax RightMain = createFollowerSparkMAXInverted(kElevatorRightMain);
+    public CANSparkMax RightSecondary = createFollowerSparkMAXInverted(kElevatorRightSecondary);
+
+    // Check if limit switch is forward or Reverse irl
+    // Check if we are going to use Normally closed or Normally open
+    // used to sync with encoders
+    public SparkMaxLimitSwitch limitSwitchLeftMain = LeftMain.getForwardLimitSwitch(Type.kNormallyClosed);
+
+    // Creates an incremental encoder
+    public RelativeEncoder encoder = LeftMain.getAlternateEncoder(SparkMaxAlternateEncoder.Type.kQuadrature, 8192);
+    // Method that creates and sets up a PID controller
+    public SparkMaxPIDController createAndSetupPIDController(CANSparkMax motor) {
+        SparkMaxPIDController PIDController = motor.getPIDController();
+        PIDController.setFeedbackDevice(encoder);
+        double kP = 0.1;
+        double kI = 1e-4;
+        double kD = 1;
```

Comment:
Should kI and KD be 0 before you start tuning?  I think this controller's suppose to get the motor to a particular angle.  I think a Kp on its own might work.

Reply:
Thanks did not know this